app/models/contact_detail.rb

```diff
@@ -0,0 +1,59 @@
+class ContactDetail
+  include ActiveModel::Model
+  attr_accessor :trainee
+
+  FIELDS = %w[
+    locale_code
+    address_line_one
+    address_line_two
+    town_city
+    postcode
+    international_address
+    phone_number
+    email
+  ].freeze
+
+  delegate :id, :persisted?, to: :trainee
+
+  attr_accessor(*FIELDS)
+
+  validates :locale_code, presence: true
+  validate :international_address_not_empty
+  validate :uk_address_must_not_be_empty
+  validates :postcode, postcode: true
+  validates :phone_number, presence: true
+  validates :phone_number, phone: true
+  validates :email, presence: true
+  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }
+
+  def initialize(trainee:)
+    @trainee = trainee
+    super(trainee.attributes.slice(*FIELDS))
+  end
+
+private
+
+  def international_address_not_empty
+    return unless trainee.non_uk?
+
+    if trainee.international_address.blank?
+      errors.add(
+        :international_address,
+        :blank,
+      )
+    end
+  end
+
+  def uk_address_must_not_be_empty
+    return unless trainee.uk?
+
+    %w[address_line_one town_city postcode].each do |field|
```

Comment:
Suggest putting `%w[address_line_one town_city postcode]` into a constant to avoid new instances getting created every time the method gets called.

Reply:
Addressed in #9fab792